### PR TITLE
Clarify the meaning of Stretch Mode/Aspect/Shrink

### PR DESCRIPTION
**Pull request by ttencate**
... using some animated GIFs and more consistent terminology.

### CODE REVIEW

**Comment by cbscribe** [comment]
This is sorely needed. I will read through, but this is a great idea, especially the animated gifs.

**Comment by ttencate** [comment]
Thank you. Please be thorough :) This needs to be good because it seems to be a frequent stumbling block for newcomers.

File: tutorials/viewports/multiple_resolutions.rst [hunk 1]
```diff
@@ -3,21 +3,24 @@
...
 
-Base resolution
----------------
+Base size
+---------
 
-A base screen resolution for the project can be specified in the project
-settings under ``display``, ``window``
+A base size for the project can be specified in the project settings
+under "Display", "Window".
```
Comment (cbscribe) [inline tutorials/viewports/multiple_resolutions.rst]: There's not a hard "standard", but most of the rest of the docs use **Project Settings** for the settings window and "Display/Window" to refer to the subsections to distinguish where to go.

File: tutorials/viewports/multiple_resolutions.rst [hunk 2]
```diff
@@ -3,21 +3,24 @@
...
 ====================
 
-Base resolution
----------------
+Base size
+---------
 
-A base screen resolution for the project can be specified in the project
-settings under ``display``, ``window``
+A base size for the project can be specified in the project settings
```
Comment (cbscribe) [inline tutorials/viewports/multiple_resolutions.rst]: I think "screen size" or "window size" would sound better. "size for the project" is a little too generic.

File: tutorials/viewports/multiple_resolutions.rst [hunk 3]
```diff
@@ -3,21 +3,24 @@
...
-it fails). On mobile, consoles or devices with a fixed resolution or
-full screen rendering, this resolution will be ignored and the native
-resolution will be used instead. To compensate for this, Godot offers
-many ways to control how the screen will resize and stretch to different
-screen sizes.
+However, what it does is not completely obvious; the engine will *not*
+attempt to switch the monitor to this resolution. Rather, think of this
+setting as the "design size", i.e. the size of the area that you work
+with in the editor. This setting corresponds directly to the size of the
+magenta rectangle in the 2D editor.
```
Comment (cbscribe) [inline tutorials/viewports/multiple_resolutions.rst]: The window size box in 2D view is blue. It's only magenta along the top where it overlaps with the red x-axis.

File: tutorials/viewports/multiple_resolutions.rst [hunk 4]
```diff
@@ -3,21 +3,24 @@
...
+
+Of course, there is often a need to support devices with screen sizes
```
Comment (cbscribe) [inline tutorials/viewports/multiple_resolutions.rst]: Remove "Of course"

File: tutorials/viewports/multiple_resolutions.rst [hunk 5]
```diff
@@ -3,21 +3,24 @@
...
+and window size that are different from this base size. Godot offers
```
Comment (cbscribe) [inline tutorials/viewports/multiple_resolutions.rst]: "screen and window sizes"

File: tutorials/viewports/multiple_resolutions.rst [hunk 6]
```diff
@@ -3,21 +3,24 @@
...
+many ways to control how the viewport will be resized and stretched to
```
Comment (cbscribe) [inline tutorials/viewports/multiple_resolutions.rst]: resized or stretched

File: tutorials/viewports/multiple_resolutions.rst [hunk 7]
```diff
@@ -44,52 +47,144 @@ of configuration variables that provide several options:
...
-   be stretched horizontally (and more horizontal resolution will be
-   reported in the viewport, proportionally). This is usually the best
-   option for 2D games that scroll horizontally (like runners or
-   platformers).
--  **Expand**: Keep aspect ratio when stretching the screen, but keep
-   neither width nor height. Depending on the screen aspect ratio, the
-   viewport will either report more horizontal resolution (if the screen
-   is wider than the original resolution) or more vertical resolution
-   (if the screen is taller than the original resolution).
+The animations below use a "base size" of just 16×9 pixels to
```
Comment (cbscribe) [inline tutorials/viewports/multiple_resolutions.rst]: remove "just"

File: tutorials/viewports/multiple_resolutions.rst [hunk 8]
```diff
@@ -44,52 +47,144 @@ of configuration variables that provide several options:
...
+16×9 pixels in size, covers the entire viewport, and a diagonal
+:ref:`Line2D <class_Line2D>` is added on top of it:
+
+.. image:: img/stretch_demo_scene.png
+
+.. Animated GIFs are generated from:
+.. https://github.com/ttencate/godot_scaling_mode
+
+-  Stretch Mode = **Disabled** (default): No stretching happens. One unit
+   in the scene corresponds 1:1 to one pixel on the screen. In this
```
Comment (cbscribe) [inline tutorials/viewports/multiple_resolutions.rst]: remove "1:1" as it's redundant

File: tutorials/viewports/multiple_resolutions.rst [hunk 9]
```diff
@@ -44,52 +47,144 @@ of configuration variables that provide several options:
...
+   pixels and screen pixels, which may result in scaling artifacts.
+   
+   This is a good option if your 2D artwork has a sufficiently high
+   resolution and does not require pixel-perfect rendering. Consider
+   enabling texture filtering and mipmapping on your 2D textures and
+   fonts.
+
+   .. image:: img/stretch_2d_expand.gif
+
+-  Stretch Mode = **Viewport**: Viewport scaling means that size of the
```
Comment (cbscribe) [inline tutorials/viewports/multiple_resolutions.rst]: "the size"

File: tutorials/viewports/multiple_resolutions.rst [hunk 10]
```diff
@@ -44,52 +47,144 @@ of configuration variables that provide several options:
...
+   in the vertical direction (if the screen is taller than the original
+   size).
+
+   .. image:: img/stretch_viewport_expand.gif
+
+Stretch Shrink
+^^^^^^^^^^^^^^
+
+The "Shrink" setting allows you to add an extra scaling factor on top of
+what the "Stretch" options above already provide. The default, 1, means
```
Comment (cbscribe) [inline tutorials/viewports/multiple_resolutions.rst]: "The default value of ``1`` means

File: tutorials/viewports/multiple_resolutions.rst [hunk 11]
```diff
@@ -44,52 +47,144 @@ of configuration variables that provide several options:
...
+that no scaling occurs.
+
+If you set "Shrink" to, say, 4 and leave "Stretch Mode" on "Disabled",
```
Comment (cbscribe) [inline tutorials/viewports/multiple_resolutions.rst]: If, for example, you set "Shrink" to ``4``

File: tutorials/viewports/multiple_resolutions.rst [hunk 12]
```diff
@@ -44,52 +47,144 @@ of configuration variables that provide several options:
...
+Stretch Aspect
+^^^^^^^^^^^^^^
+
+The second setting is the stretch aspect. Note that this only takes effect if
+"Stretch Mode" is set to something other than "disabled".
+
+In the animations below, you will notice gray and black areas. The black
+areas are added by the engine and cannot be drawn into. The gray areas
+are part of your scene, and can be drawn to. The gray areas correspond
+to the region outside the magenta frame you see in the 2D editor.
```
Comment (cbscribe) [inline tutorials/viewports/multiple_resolutions.rst]: magenta -> blue

**Comment by mhilbrunner** [comment]
Thanks, this is a big improvement :)

**Pull request merged**